src/reactThisBindingIssueRule.ts

```diff
@@ -68,9 +69,9 @@ class ReactThisBindingIssueRuleWalker extends ErrorTolerantWalker {
         this.getOptions().forEach((opt: any) => {
             if (typeof(opt) === 'object') {
                 this.allowAnonymousListeners = opt['allow-anonymous-listeners'] === true;
-            }
-            if (typeof opt === 'string') {
-                this.allowedDecorators = this.allowedDecorators.concat(opt);
+                if (opt['bind-decorators']) {
```

Comment:
I think it might be safer to check it's an array before assigning it to `allowedDecorators`. Maybe even filter to string only.